azure-pipelines/publishing/github-release.yml

```diff
@@ -9,96 +9,122 @@ pr:
+        targetOsArch: ubuntu1804-arm64
+        dockerImageName: 'jimsonmsft/ubuntu18.04-arm64:latest'
+        config: release
 
-  - task: DownloadBuildArtifacts@0
-    inputs:
-      buildType: 'specific'
-      project: '7cdf52bf-a6f7-436c-b12f-3f063ef389c4'
-      pipeline: '26'
-      buildVersionToDownload: 'latest'
-      downloadType: 'specific'
-      downloadPath: '/tmp/do_publishing'
-    displayName: 'Get ms-do-sdk x64 artifacts'
+  - job: libdeliveryoptimization
+    steps:
+    - template: ../build/templates/dosdkcpp-native-steps.yml
+      parameters:
+        targetOsArch: 'ubuntu1804-x64'
+        config: release
+        skipTests: true 
+    - template: ../build/templates/dosdkcpp-docker-steps.yml
+      parameters:
+        targetOsArch: debian10-arm32
+        dockerImageName: 'jimsonmsft/debian10-arm32:latest'
+        config: release
+    - template: ../build/templates/dosdkcpp-docker-steps.yml
+      parameters:
+        targetOsArch: debian9-arm32
+        dockerImageName: 'jimsonmsft/debian9-arm32:latest'
+        config: release
+    - template: ../build/templates/dosdkcpp-docker-steps.yml
+      parameters:
+        targetOsArch: ubuntu1804-arm64
+        dockerImageName: 'jimsonmsft/ubuntu18.04-arm64:latest'
+        config: release
 
-  - task: DownloadBuildArtifacts@0
-    inputs:
-      buildType: 'specific'
-      project: '7cdf52bf-a6f7-436c-b12f-3f063ef389c4'
-      pipeline: '25'
-      buildVersionToDownload: 'latest'
-      downloadType: 'specific'
-      downloadPath: '/tmp/do_publishing'
-    displayName: 'Get ms-do-agent arm artifacts'
+  - job: deliveryoptimization_plugin_apt
+    steps:
+    - template: ../build/templates/dopapt-native-steps.yml
+      parameters:
+        targetOsArch: 'ubuntu1804-x64'
+        config: release
+    - template: ../build/templates/dopapt-docker-steps.yml
+      parameters:
+        targetOsArch: debian10-arm32
+        dockerImageName: 'jimsonmsft/debian10-arm32:latest'
+        config: release
+    - template: ../build/templates/dopapt-docker-steps.yml
+      parameters:
+        targetOsArch: debian9-arm32
+        dockerImageName: 'jimsonmsft/debian9-arm32:latest'
+        config: release
+    - template: ../build/templates/dopapt-docker-steps.yml
+      parameters:
+        targetOsArch: ubuntu1804-arm64
+        dockerImageName: 'jimsonmsft/ubuntu18.04-arm64:latest'
+        config: release
 
-  - task: DownloadBuildArtifacts@0
-    inputs:
-      buildType: 'specific'
-      project: '7cdf52bf-a6f7-436c-b12f-3f063ef389c4'
-      pipeline: '23'
-      buildVersionToDownload: 'latest'
-      downloadType: 'specific'
-      downloadPath: '/tmp/do_publishing'
-    displayName: 'Get ms-do-agent x64 artifacts'
+- stage: release
+  condition: succeeded()
+  jobs:
+  - job: github_release
+    condition: eq(${{parameters.SkipPublishing}}, false)
+    steps:
+    - task: DownloadBuildArtifacts@0
+      inputs:
+        buildType: 'current'
+        downloadType: 'specific'
```

Comment:
>specific [](start = 23, length = 8)

Did you intend to specify an itemPattern to download only .deb files?

My suggestion would be to download all artifacts so that we can do things like calculate and publish hashes of each artifact (later).

Reply:
No, the default here is single.

The only other option is 'specific'
https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/utility/download-build-artifacts?view=azure-devops

I intend to download all artifacts, my guess was 'single' was not the right option to do so.
We'll see when the current build completes whether or not 'specific' downloads all artifacts.

Reply:
I think you're right.

---
In reply to: [590816853](https://github.com/microsoft/do-client/pull/7#discussion_r590816853) [](ancestors = 590816853)